crates/rolldown/tests/esbuild/default/var_relocating_no_bundle/artifacts.snap

```diff
@@ -11,8 +11,6 @@ source: crates/rolldown_testing/src/integration_test.rs
 function x() {
 	{
 		var a;
-		for (var b; 0;);
-		for (var { c, x: [d] } = {}; 0;);
```

Comment:
This is incorrect.

We need to handle it to:
```js
var{c:a,x:[c]}={};
```

Reply:
this is the original code: 
```js
	for (var { c, x: [d] } = {}; 0;);
	```
	I don't know why need to transform it to 
	```js
	var{c:a,x:[c]}={};
	```

Reply:
I am sorry for copying the mangled version.

The assignment can be side-effecting, yet we don't keep it.

For example, for the code:

```
for (var { c, x: [d] } = f(); 0;);
```

We still eliminate it.

I am also investigating the destruction, but yet I don't have enough bandwidth these days.

Reply:
Yeah, the esbuild considering the code as side effects

Reply:
result of swc: 
https://play.swc.rs/?version=1.7.35&code=H4sIAAAAAAAAA%2BNMyy9S0ChLLFKoVkjWUaiwUohOiVWoVbBVqK61VjCw1rTmAgCuG4QNIwAAAA%3D%3D&config=H4sIAAAAAAAAA31VuXLjMAzt8xUe1Zn1josUaXcn3Vb7ARxaBCVkRUJDgI49Gf%2F7Qodvyp3Nh4cbT98vq1X1yXX1vvrWn%2Fqnt4khnf%2FrCx%2Bi2L2%2BVHLogeuEvVSvJ1R4gCRlGF%2BOE1CJTQ3IQALe%2FNxsqtN7spE9pXAdITP8Bo8RPij96izzB0LnePZ7DpXA1nLF0ycMPSX5SznVMARrRXp%2BX6%2BBww9u1%2F1Eeb0QUo6CYbS1WShYwbqa4eNNBR0RD3bedjznUAWM6A%2FXqdcUNAjzbVpafA4QhW%2F5M5bo6742fd8SdaDNWUaMZYNRoIFUclxT19mewexsKngZMrUJmUohBjALONMn6ot4dChIUWM%2Bog6sMzU5KECYoBbcQYmmsZQWWcsr1DPCDra5acaFvGPDznbZSiEm7MeRaLYFry0hi%2FE5llo4gQs9mMC5ufdM9CaB5BQfeZ%2BEcWEm%2FwC0A8PGRxug5He08LpPS2z%2FlIlRzwrlUMB1v0tVRmi0qQbRFzo7dAaSYGmaCVyuYehsXUpnhhfax%2BjAgPe6KwXX%2FIVSt6WggyKRLwA6X%2BtLWzUB5nyFC%2FhwEE%2FgD61Sygs2W6iytMsoH8KWuicBAkhL7omBjkJoGU6qEvt%2BGc%2FRjYrriiYqyK4oAnoAQqYblf1hN%2FQ81KNpOtpeZOKkrWcVDzY2l3ufJPdlNqgCuTyC89domO%2F0EXmrLkYnFT5lUCH%2FORHHmMf%2Fs1GtlNgGAAA%3D

Reply:
Maybe I can hoist the `init` instead of the elimination.

Reply:
Let's rerun the CI. Related bugs are fixed: https://github.com/oxc-project/oxc/issues/6547.